src/pybind/mgr/volumes/fs/volume.py

```diff
@@ -132,14 +132,20 @@ def create_subvolume(self, **kwargs):
         volname    = kwargs['vol_name']
         subvolname = kwargs['sub_name']
         groupname  = kwargs['group_name']
+        size       = kwargs['size']
+        pool       = kwargs['pool_layout']
+        uid        = kwargs['uid']
+        gid        = kwargs['gid']
 
         try:
             with open_volume(self, volname) as fs_handle:
                 with open_group(fs_handle, self.volspec, groupname) as group:
                     try:
-                        with open_subvol(fs_handle, self.volspec, group, subvolname):
-                            # idempotent creation -- valid.
-                            pass
+                        with open_subvol(fs_handle, self.volspec, group, subvolname) as subvolume:
+                            # idempotent creation -- valid. Attributes set is supported.
+                            uid = uid if uid else subvolume.uid
+                            gid = gid if gid else subvolume.gid
+                            subvolume._set_attrs(subvolume.path, size, False, pool, uid, gid)
```

Comment:
Also '_set_attrs' method can lose the leading underscore. 
https://stackoverflow.com/a/1301557

Reply:
> Why is isolate_namespace always set to `False`? You are allowing pool layout to be set, but why not pool namespace?

The PR https://github.com/ceph/ceph/pull/34776 is not merged yet. I will rebase once that is merged.

Reply:
> Also '_set_attrs' method can lose the leading underscore.
> https://stackoverflow.com/a/1301557

But the test is passing. Also python doesn't enforce private/protected concepts explicitly, I think. So, it's working fine. Do you suggest to remove the leading underscore for that function since it's getting accessed from the object outside the class ?

Reply:
>  Do you suggest to remove the leading underscore for that function since it's getting accessed from the object outside the class ?

yes

Reply:
> Also python doesn't enforce private/protected concepts explicitly, I think.

Yes, it only differs if you use a wildcard import. It's just a naming convention otherwise, afaik.

Reply:
Done